### PR TITLE
Polyhedron demo : add vtk_plugin

### PR DESCRIPTION
**Pull request by janetournois**
this PR introduces Polyhedron_demo_vtk_plugin to load .vtk and .vtp files in a Scene_polyhedron_item

### CODE REVIEW

**Comment by janetournois** [comment]
merged in integration

**Comment by lrineau** [comment]
Merged in [CGAL-4.8-Ic-23](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-23.shtml).

**Comment by lrineau** [comment]
In this pull-request, you link `vtk_plugin` to all VTK modules. Well, maybe not all, but on my machine, that links with 166 different VTK modules. I suggest you use the option `COMPONENTS ...` in `find_package(VTK ...)`.

**Comment by janetournois** [comment]
you're right, I'll try to improve that
thanks

**Comment by janetournois** [comment]
I remerged this branch to integration after commmit ee331d0

**Comment by sloriot** [comment]
Maybe the name of the plugin is too broad for now. It only reads `PolyData` files.

**Comment by lrineau** [comment]
We already have too many plugins, anyway. If we extend the functionality to read more VTK file format, we will almost certainly implement that in the same plugin. If the plugin code is clear on that, that is fine.

File: Polyhedron/demo/Polyhedron/Polyhedron_demo_vtk_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+{
+  Q_OBJECT
+  Q_INTERFACES(Polyhedron_demo_io_plugin_interface)
+  Q_PLUGIN_METADATA(IID "com.geometryfactory.PolyhedronDemo.IOPluginInterface/1.0")
+
+public:
+  typedef boost::graph_traits<Polyhedron>::vertex_descriptor vertex_descriptor;
+  typedef boost::graph_traits<Polyhedron>::face_descriptor face_descriptor;
+
+  QString nameFilters() const { return "VTK files (*.vtk, *.vtp)"; }
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Polyhedron_demo_vtk_plugin.cpp]: `"VTK files"` could be changed to `"VTK PolyData files"`, to emphasis that we do not support the other data structures.

Comment (janetournois) [inline Polyhedron/demo/Polyhedron/Polyhedron_demo_vtk_plugin.cpp]: I'll make this change, it clarifies things

**Comment by lrineau** [comment]
[Compilation error here][e].

[e]: https://cgal.geometryfactory.com/CGAL/Members/testsuite/CGAL-4.8-Ic-23/Polyhedron_Demo/TestReport_lrineau_x86-64_Linux-Fedora19_g++-4.8_ansi.gz

```
In file included from /usr/include/vtk/vtkAlgorithm.h:33:0,
                 from /usr/include/vtk/vtkDataReader.h:30,
                 from /usr/include/vtk/vtkPolyDataReader.h:32,
                 from /no-raid/home/lrineau/CGAL/CGAL-4.8-Ic-23/cmake/platforms/x86-64_Linux-Fedora19_g++-4.8_ansi/test/Polyhedron_Demo/Polyhedron_demo_vtk_plugin.cpp:43:
/usr/include/vtk/vtkObject.h:53:40: error: extra ';' [-Wpedantic]
   vtkTypeMacro(vtkObject,vtkObjectBase);
                                        ^
```

**Comment by lrineau** [comment]
@janetournois: Now that `Three` is merged in `master`, you should also cherry-pick the commit f4e3b11879fee034c9034d50c5fc7ab926735b57 to your branch.

**Comment by janetournois** [comment]
remerged into [CGAL-4.8-Ic-26](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-26.shtml)

**Pull request merged**